spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfigurationTests.java

```diff
@@ -51,118 +47,102 @@
+						SecurityAutoConfiguration.class, DataFlowServerAutoConfiguration.class,
+						DataFlowControllerAutoConfiguration.class, DataSourceAutoConfiguration.class,
+						DataFlowServerConfiguration.class, PropertyPlaceholderAutoConfiguration.class,
+						RestTemplateAutoConfiguration.class, HibernateJpaAutoConfiguration.class,
+						SchedulerConfiguration.class, JacksonAutoConfiguration.class, SimpleTaskAutoConfiguration.class,
+						ResourceLoadingAutoConfiguration.class, ComposedTaskRunnerConfigurationProperties.class);
 	}
 
 	/**
 	 * Verify that embedded server starts if h2 url is specified with default properties.
 	 */
 	@Test
-	@Ignore
 	public void testStartEmbeddedH2Server() {
-		Map<String, Object> myMap = new HashMap<>();
-		myMap.put("spring.datasource.url", "jdbc:h2:tcp://localhost:19092/mem:dataflow");
-		myMap.put("spring.dataflow.embedded.database.enabled", "true");
-		propertySources.addFirst(new MapPropertySource("EnvironmentTestPropsource", myMap));
-		context.setEnvironment(environment);
-
-		context.refresh();
-		assertTrue(context.containsBean("initH2TCPServer"));
+		contextRunner.withPropertyValues(
+						"spring.datasource.url=jdbc:h2:tcp://localhost:19092/mem:dataflow",
+						"spring.dataflow.embedded.database.enabled=true")
+				.run(context -> {
+					assertTrue(context.containsBean("h2TcpServer"));
+					Server server = context.getBean("h2TcpServer", Server.class);
+					assertTrue(server.isRunning(false));
+
+					// Verify H2 Service is stopped
+					context.close();
+					assertFalse(server.isRunning(false));
+				});
 	}
 
 	/**
-	 * Verify that embedded h2 does not start if h2 url is specified with with the
+	 * Verify that embedded h2 does not start if h2 url is specified with the
 	 * spring.dataflow.embedded.database.enabled is set to false.
-	 *
-	 * @throws Throwable if any error occurs and should be handled by the caller.
 	 */
-	@Test(expected = ConnectException.class)
-	public void testDoNotStartEmbeddedH2Server() throws Throwable {
-		Throwable exceptionResult = null;
-		Map<String, Object> myMap = new HashMap<>();
-		myMap.put("spring.datasource.url", "jdbc:h2:tcp://localhost:19092/mem:dataflow");
-		myMap.put("spring.dataflow.embedded.database.enabled", "false");
-		myMap.put("spring.jpa.database", "H2");
-		propertySources.addFirst(new MapPropertySource("EnvironmentTestPropsource", myMap));
-		context.setEnvironment(environment);
-		try {
-			context.refresh();
-		}
-		catch (BeanCreationException exception) {
-			exceptionResult = exception.getRootCause();
-		}
-		assertNotNull(exceptionResult);
-		throw exceptionResult;
+	@Test
+	public void testDoNotStartEmbeddedH2Server() {
+		contextRunner.withPropertyValues(
+						"spring.datasource.url=jdbc:h2:tcp://localhost:19092/mem:dataflow",
+						"spring.dataflow.embedded.database.enabled=false",
+						"spring.jpa.database=H2"
+				)
+				.run(context -> {
+					assertNotNull(context.getStartupFailure());
```

Comment:
There is a rich assert model you can use for startup failures :
```java
					assertThat(context).hasFailed()
							.getFailure()
							.isInstanceOf(BeanCreationException.class)
							.hasMessage("....");
```
There are many `hasMessage**` variants to use etc..